gcc/common/config/riscv/riscv-common.c

```diff
@@ -645,6 +651,19 @@ static const riscv_ext_flag_table_t riscv_ext_flag_table[] =
   {"f", &gcc_options::x_target_flags, MASK_HARD_FLOAT},
   {"d", &gcc_options::x_target_flags, MASK_DOUBLE_FLOAT},
   {"c", &gcc_options::x_target_flags, MASK_RVC},
+
+  {"k", &gcc_options::x_riscv_crypto_subext, MASK_ZKNE | MASK_ZKND | MASK_ZKNH | MASK_ZKG | MASK_ZKB | MASK_ZKR},
```

Comment:
I would suggest you using `riscv_implied_info` to record the implied extension info like `zb*`, you could reference her e: https://github.com/riscv/riscv-gcc/blob/riscv-gcc-10.2.0-rvb/gcc/common/config/riscv/riscv-common.c#L63